src/views/HomeView.vue

```diff
@@ -1,32 +1,172 @@
 <template>
-  <div id="greetings">
-    <h1>{{ message }}</h1>
+  <div class="Container">
+    <img src="@/assets/images/logo-and-lettering.svg" alt="Saci Logo" class="Logo" />
+    <h1>Choose the role you want</h1>
+    <div class="Dropdown">
+      <ElSelect v-model="localSelectedRole" placeholder="Select a role" size="large">
+        <ElOption
+          v-for="item in rolesStore.roles"
+          :key="item.name"
+          :label="item.label"
+          :value="item.name"
+        />
+      </ElSelect>
+    </div>
+  </div>
+
+  <div v-if="role && role.knowledges && role.knowledges.length" class="Container">
+    <h2>
+      Now, level the ability you think you <br />
+      have in each technology below. 0 you <br />know nothing, and 5 you master the knowledge
+    </h2>
+    <div v-for="(knowledge, index) in role.knowledges" :key="index" class="KnowledgeItem">
+      <div class="KnowledgeName">{{ knowledge.name }}</div>
+      <div class="KnowledgeInputs">
+        <ElSlider
+          v-model="quizStore.quizAnswers.answers[index].answer"
+          :min="0"
+          :max="5"
+          :step="1"
+          show-stops
+        />
+      </div>
+    </div>
+    <ElButton type="primary" @click="getResult()">Get Result</ElButton>
+  </div>
+
+  <div v-if="quizStore.quizResult" class="ResultScreen">
+    <img src="@/assets/images/logo-and-lettering.svg" alt="Saci Logo" class="Logo" />
+    <p>
+      Considering your responses, you are considered<br />
+      a <strong>{{ quizStore.quizResult.currentLevel?.name }}</strong
+      >!
+    </p>
+    <p v-if="quizStore.quizResult.nextLevel">
+      If you want to evolve technically, click
+      <a :href="quizStore.quizResult.nextLevel.link">here</a> to get access<br />
+      to a technical knowledge path
+    </p>
+    <p v-else>You are currently at the highest level.</p>
   </div>
 </template>
 
 <script lang="ts" setup>
-import { ref, onMounted } from 'vue'
+import { ref, onMounted, watch } from 'vue'
+import { ElOption, ElSelect, ElSlider, ElButton } from 'element-plus'
+import { useRolesStore } from '@/stores/roles'
+import { useQuizStore } from '@/stores/quiz'
+import type { Role } from '@/domain/Role'
+import { ErrorMessage, SuccessMessage } from '@/services/messages'
+
+const rolesStore = useRolesStore()
+const quizStore = useQuizStore()
 
-const message = ref('Loading...')
+const role = ref<Role | null>(null)
+const localSelectedRole = ref<string | null>(null)
 
-const fetchMessage = async (): Promise<string> => {
+onMounted(async () => {
+  await rolesStore.fetchRoles()
+  if (rolesStore.roles.length > 0) {
+    localSelectedRole.value = ''
+    handleRoleChange(rolesStore.roles[0])
+  }
+})
+
+watch(localSelectedRole, (newRoleName) => {
```

Comment:
You don't need this watch. There is an event called @change for select. Check in  element-plus documentation. Then you don't need the localSelectedRole neither